// nmea2decimal/nmea2decimal.js
exports.Location =  {
    WEST:   'W',
    EAST:   'E',
    NORTH:  'N',
    SOUTH:  'S'
};

var nmea2decimal = {

    latitudeToDecimal: function (latitude, pos) {
        if (typeof latitude == "undefined") {
            return null;
        }

        var decimal = parseFloat(latitude.substring(0,2)) + parseFloat(latitude.substring(2)) / 60;

        if (decimal > 0 && pos == exports.Location.WEST || pos == exports.Location.SOUTH) {
            return decimal * -1;
        } else {
            return decimal;
        }
    },

    longitudeToDecimal: function (longitude, pos) {
        if (typeof longitude == "undefined") {
            return null;
        }

        var decimal = parseFloat(longitude.substring(0,3)) + parseFloat(longitude.substring(3)) / 60;

        if (decimal > 0 && pos == exports.Location.WEST || pos == exports.Location.SOUTH) {
            return decimal * -1;
        } else {
            return decimal;
        }
    },

    getDecimal: function (latitude, latitudePos, longitude, longitudePos) {
        return {
            latitude: this.latitudeToDecimal(latitude, latitudePos),
            longitude: this.longitudeToDecimal(longitude, longitudePos)
        };
    }

};

module.exports = nmea2decimal;
